keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1655 @@
+  because they modify containers and that works on all nodes at all
+  times. Forcing such drivers to set an allow list would cause unnecessary
+  work.
+
+In its state for the Pod the scheduler plugin must remember when it rejected a
+node because of UnsuitableNodes. That information will be used in Post-filter
+to deallocate resources.
+
+#### Post-filter
+
+This is called when no suitable node could be found. If the Pod depends on ResourceClaims with delayed
+allocation, then deallocating one or more of these ResourceClaims may make the
+Pod schedulable after allocating the resource elsewhere. Therefore each
+ResourceClaim with delayed allocation is checked whether all of the following
+conditions apply:
+- allocated
+- not currently in use
+- it was the reason why some node could not fit the Pod, as recorded earlier in
+  Filter
+
+One of the ResourceClaims satisfying these criteria is picked randomly and its
+Phase is set to Deallocate. This may make it possible to run the Pod
+elsewhere. If it still doesn't help, deallocation may continue with another
+ResourceClaim, if there is one. To prevent deallocating all resources merely
+because the scheduler happens to check quickly, the next deallocation will only
+requested when there is none currently running.
+
+At the moment, the scheduler has no information that might enable it to
+prioritize which resource to deallocate first. Future extensions of this KEP
+might attempt to improve this.
+
+#### Pre-score
+
+This is passed a list of nodes that have passed filtering by the resource
+plugin and the other plugins. The PotentialNodes field of unallocated
+ResourceClaims with delayed allocation gets updated now if the field doesn't
+match the current list already.
+
+#### Reserve
+
+A node has been chosen for the Pod.
+
+If using delayed allocation and the resource has not been allocated yet,
+the SelectedNode field of the ResourceClaim
+gets set here and the scheduling attempt gets stopped for now. It will be
+retried when the ResourceClaim status changes.
+
+If all resources have been allocated already,
+the scheduler adds the Pod to the ReservedFor field of its ResourceClaims to ensure that
+no-one else gets to use those.
+
+If some resources are not allocated yet or reserving an allocated resource
+fails, the scheduling attempt needs to be aborted and retried at a later time
+or when the ResourceClaims change.
+
+#### Unreserve
+
+The scheduler removes the Pod from the ReservedFor field because it cannot be scheduled after
+all.
+
+### kubelet
+
+#### Managing resources
+
+kubelet must ensure that resources are ready for use on the node before running
+the first Pod that uses a specific resource instance and make the resource
+available elsewhere again when the last Pod has terminated that uses it. For
+both operations, kubelet calls a resource kubelet plugin as explained in the next
+section.
+
+Pods that are not listed in ReservedFor or where the ResourceClaim doesn't
```

Comment:
This means that kubelets are list/watching all ResourceClaim, and from the enumerated type def, it looks like those objects are potentially very large with a number of unbounded lists.  This represents a significant performance risk.

Reply:
Would it help to get the KEP accepted if I add a field that kubelet can filter on?